compiler/src/main/java/com/squareup/anvil/compiler/ModuleMerger.kt

```diff
@@ -173,6 +188,20 @@ internal class ModuleMerger(
     block(this)
     visitEnd()
   }
+
+  private fun createAnvilModuleName(classDescriptor: ClassDescriptor): FqName {
+    val name = "$MODULE_PACKAGE_PREFIX.${classDescriptor.findPackage().fqName}." +
+        classDescriptor.parentsWithSelf
+            .filterIsInstance<ClassDescriptor>()
+            .toList()
+            .reversed()
```

Comment:
I saw this in your last PR, but still curious why you need to reverse this.

Reply:
Explained it there.